### DISCUSSION TITLE
No-cache loaders

### DISCUSSION THREAD

**Discussion by tel** [observablehq/framework#1848] [Ideas]
For example, a loader which just calls `ls` on a directory. It's very cheap, so caching doesn't help much. At the same time, if that directory changes the cached loader value will persist requiring the user to manually delete the cached version.

As a proposal, inspired by Rust's `cargo`, perhaps directives could be emitted on stderr, like `!observable.loader.no_cache=true`. This opens up design space for other runtime loader metadata, too. I could imagine a loader explicitly noting what it's dependencies are so it could be cache busted more effectively, too.

**Comment by Fil** [comment]
So you need to continually watch the output of that command, right? Maybe you could do this with a command such as `watch "ls > src/data/ls.txt"`… with proper parameters to avoid reloading and recomputing server-side each time it runs even if the output is unchanged.

Also does this mean you are working exclusively with the preview server?

**Reply by tel** [reply]
Currently, yes. I am regularly regenerating new experimental results and reviewing them using the report.md I've built. I don't have much use at the moment for fully static builds.

I can definitely use a workaround like `watch`. It just adds to the server setup.
